Connect4/Connect4_hardcore.py

```diff
@@ -0,0 +1,268 @@
+# Additions in hardcore mode:
+# Customizable board size
+# Option to play new game
+# Win counter
+
+board_columns = 7
+board_rows = 6
+board = []
+players = ["r", "y"]
+current_player = players[0]
+symbols = {
+    "r": "🔴",
+    "y": "🟡",
+    " ": "⚪",
+    0: "0️⃣ ",
+    1: "1️⃣ ",
+    2: "2️⃣ ",
+    3: "3️⃣ ",
+    4: "4️⃣ ",
+    5: "5️⃣ ",
+    6: "6️⃣ ",
+    7: "7️⃣ ",
+    8: "8️⃣ ",
+    9: "9️⃣ ",
+}
+wins = {
+    "r": 0,
+    "y": 0
+}
+
+
+def print_board():
+    """Print out the board."""
+
+    # adds newline before displaying the board
+    print()
+
+    # the below code prints out each digit of the column indexes in a seperate line,
+    # so if a board size wider than 10 is chosen, 
+    # the indexes don't offset due to them having multiple digits
+
+    # repeats for each digit in the largest board column index
+    for digit in range(len(str(len(board[0])))):
+        # creates string that will display the specific digit of each index
+        column_indexes_print = ""
+
+        # repeats for each board column
+        for column_index in range(len(board[0])):
+
+            # pads the column index to be the same length as the largest index
+            # so if the index doesn't have a tens digit for example, it won't print anything
+            padded_column_index = str(column_index).rjust(len(str(len(board[0]))), "n")
+            
+            # check if the index has the specific digit
+            # if yes, add the digit (converted to emoji form) to column_indexes_print
+            # if no, adds an empty space to column_indexes_print
+            if padded_column_index[digit].isdigit():
+                column_indexes_print += symbols[int(padded_column_index[digit])]
+            else:
+                column_indexes_print += "  "
+
+        # print out the specific digit of each column index
+        print(column_indexes_print)
+
+    # repeats for each row in the board
+    for row in range(len(board)):
+        # creates string that will represent the values in the row
+        row_print = ""
+
+        # adds each value in the row to row_print
+        for space in board[row]:
+            row_print += symbols[space]
+
+        # prints the values in the row
+        print(row_print)
+
+
+def bottom_row(column):
+    """Return the bottommost empty space in a column."""
+    for row in reversed(range(len(board))):
+        if board[row][column] == " " and column >= 0:
+            return row
+
+    return None
+
+
+def switch_player():
+    """Switch the current player"""
+    global current_player
+    if current_player == players[0]:
+        current_player = players[1]
+    else:
+        current_player = players[0]
+
+
+def all_same(list):
+    """Take in a list, return whether all its elements are the same."""
+    for item in list:
+        if item != list[0] or item == " ":
+            return False
+
+    return True
```

Comment:
Smaller and optional nitpick. It might be better to use a different name for the `list` parameter, since the `list()` is used to construct a list/convert iterables into a list.

In addition, `all_same(list)` function can be rewritten into one line: `return list.count('r') == 4 or list.count('y') == 4`, but like it doesn't really make a difference, just shorter.

Reply:
The `list` parameter thing would also apply to normal version too.